include/circt/Dialect/Calyx/CalyxControl.td

```diff
@@ -258,3 +258,20 @@ def WhileOp : CalyxContainer<"while", [
   ];
 
 }
+
+def InvokeOp : CalyxOp<"invoke", [
+                        ControlLike,
+                        SameVariadicOperandSize
+                        ]> {
+  let summary = "Calyx Invoke";
+  let arguments = (ins FlatSymbolRefAttr:$callee, 
+                   Variadic<AnyType>:$ports,
+                   Variadic<AnyType>:$inputs);
```

Comment:
What's the difference between ports and inputs?

Reply:
The `ports` represent the `input ports` of the instance, such as a register.The `ports` are `%r.in %r.clk %r.reset`.`ports` indicates the port to `be assigned`.
```
 %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1
```
`inputs` specifies the value for there `ports`.

Reply:
Okay, then you also need some mechanism to connect the output ports defined by the invoke in the same way

Reply:
> Okay, then you also need some mechanism to connect the output ports defined by the invoke in the same way

We have `ports` and `inputs`, we just need to iterate over them and then apply `calyx.assign` to each port in turn. Of course, I did a lot of validation on `calyx.invoke`.

Reply:
Can you add an example with `calyx.invoke` being called on another `calyx.component` please with input and output ports? AFAIK, we don't `invoke` members of Calyx's standard library, e.g., registers and adders. So, those should be replaced with a new component (correct me if I'm wrong Rachit, it has been a while). In any case, a test with a new component will be useful to understand.